aws/run-on-ec2.py

```diff
@@ -137,17 +146,30 @@ def getPowermixingCommands(runid, s3Manager, instanceIds, instanceIps):
     setupCommands = []
     for i, instanceId in enumerate(instanceIds):
         commands = [
-            "sudo docker pull %s" % (AwsConfig.DOCKER_IMAGE_PATH),
             "mkdir -p config",
             "cd config; curl -sSO %s" % (instanceConfigUrl),
             "mkdir -p sharedata",
+            "git clone https://github.com/lu562/upload-script.git",
+            "cp upload-script/Download_input.sh sharedata/Download_input.sh ",
             "mkdir -p benchmark",
+            "ulimit -n 10000",
         ]
+        threads = []
         for j in range(k):
             fname = f"{powersPrefix}_{runid}_{j}-{i}.share"
-            url = s3Manager.uploadFile(fname)
-            commands.append(f"cd sharedata; curl -sSO {url}")
+            threads.append(threading.Thread(target=uploadFile, args=(fname,)))
+        for j in range(k):
+            threads[j].start()
+        for j in range(k):
+            threads[j].join()
+        with open('%s-%d-links' % (runid, i), 'w') as f:
+            while not q.empty():
+                print(q.get(), file=f)
+        fname = f"{runid}-{i}-links"
+        url = s3Manager.uploadFile(fname)
+        commands.append(f"cd sharedata; curl -sSO {url}; sh Download_input.sh {fname}")
```

Comment:
This command will have to be changed now, right?

Reply:
so right now the script is in docker image but how do we get it outside docker? since the first step is that we need to prepare inputs outside the docker container and then pass inputs into container.

Reply:
If you look at the docker commands, we do a -v x:y which mounts the folder outside the container to inside the container.

Reply:
I see, so we do not need to upload/download via s3, we can just copy to `$x/sharedata` where `$x` is the host location of the mounted volume

Reply:
I guess we still need to first use s3 to upload inputs to ec2 first, then mount inputs into containers. My question was our script for downloading input is inside container and we need to extract it out so that we can download input on ec2. And smkuls recommends also uploading scripts using s3.